src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/unit/uom/service/impl/UOMServiceImpl.java

```diff
@@ -361,31 +363,90 @@ public void updateExistingUOMByCode(String code, UOMDtoV2 sourceUOMDto) {
+            log.debug("adding all linkedUOMs");
+            List<UnitClassSelfLinkageContract> selfLinkageContracts = new ArrayList<>(newLinkedUOMs);
             selfLink(target, Optional.of(selfLinkageContracts));
-            *//*Collections.binarySearch(target.getFromUOMs(), UOMSelfLinkageEntity.builder().to.build())
-            linkedUOMsLeft.stream().map(f -> target.getFromUOMs(). f.getCode().compareTo())
-            // directly assign to UOMEntity because no comparison is needed
-            linkedUOMsLeft.stream().forEach(f -> {
-                Diff dtoDiff = javers.compare(UnitClassSelfLinkageDtoV2.builder().build(), f);
-                dtoDiff.getChangesByType(ValueChange.class).forEach(Errors.rethrow().wrap(p -> {
-                    BeanUtil.pojo.setProperty(target, "", p.getRight());
-                }));
-            });*//*
-
-        } */else if(old.getLinkedUOMs().isPresent() && _new.getLinkedUOMs().isPresent()) {
-            List<UnitClassSelfLinkageDtoV2> oldLinkedUOMDto = old.getLinkedUOMs().get();
-            List<UnitClassSelfLinkageDtoV2> newLinkedUOMDto = _new.getLinkedUOMs().get();
-            Diff diffCollections = javers.compareCollections(oldLinkedUOMDto, newLinkedUOMDto, UnitClassSelfLinkageDtoV2.class);
+
+        }
+        else if(oldLinkedUOMs.size() > 0 && newLinkedUOMs.size() > 0) {
+
+            Diff diffCollections = javers.compareCollections(oldLinkedUOMs, newLinkedUOMs, UnitClassSelfLinkageDtoV2.class);
             log.debug(diffCollections.prettyPrint());
+            diffCollections.getChangesByType(ValueChange.class).forEach(Errors.rethrow().wrap(p -> {
+
+                if(!(p instanceof InitialValueChange || p instanceof TerminalValueChange) ){
+                    if(p.getPropertyName().equals("quantity")) {
+                        log.debug("{} changed from {} to {}", p.getPropertyNameWithPath(), p.getLeft(), p.getRight());
+                        int newQuantity = Integer.parseInt(p.getRight().toString());
+                        String code = ((UnitClassSelfLinkageDtoV2) p.getAffectedObject().get()).getCode();
+                        target.getFromUOMs().forEach(e -> {
+                            if (e.getToUom().getCode().equals(code)) {
+                                e.setQuantity(newQuantity);
+                            }
+                        });
+                    }
+                }
+            }));
+
+            diffCollections.getChangesByType(NewObject.class).forEach(Errors.rethrow().wrap(p->{
+                UnitClassSelfLinkageDtoV2 elt = (UnitClassSelfLinkageDtoV2) p.getAffectedObject().get();
+                log.debug("new linkedUOM : {}", elt.getCode());
+                Optional<UOMEntity> optionalTo = uomJpaRepository.findByCode(elt.getCode());
+                if(optionalTo.isPresent()){
+                    UOMEntity to = optionalTo.get();
+                    UOMSelfLinkageEntity selfLinkage = UOMSelfLinkageEntity.builder()
```

Comment:
use bean utils to copy property from source to target instead of using property specific getters and setts as this logic needs to be dynamic for extending to other types of resources in other sub domains